### PR TITLE
Packages/libxc

### PR DESCRIPTION
**Pull request by batatasfritas**
Updated the supported versions.
Changed the URL and added individual URLs for the two versions that are different.
LibXC 4.x is needed for CP2K 6.1. So this has to be pushed upstream together/before CP2K 6.1 updates

### CODE REVIEW

**Review by rmsds — CHANGES_REQUESTED** [review]
I think you accidentally based this branch on the CP2K changes used in #199.
Could you `git rebase -i origin/releases/paien` and drop the CP2K commit?
Otherwise it looks OK.

File: var/spack/repos/builtin/packages/libxc/package.py [hunk 1]
```diff
@@ -30,11 +30,21 @@ class Libxc(AutotoolsPackage):
...
 
+    version('4.2.3', '6176ac7edf234425d973903f82199350')
+    version('4.0.3', '44d2a4873a58e6b99fb45b01571b20cb')
+    version('4.0.0', '7b987677700cebc3e30584923efd240f')
+    version('3.0.1', '7891522833953f82f3ac9118dd3bb893')
     version('3.0.0', '8227fa3053f8fc215bd9d7b0d36de03c')
-    version('2.2.2', 'd9f90a0d6e36df6c1312b6422280f2ec')
-    version('2.2.1', '38dc3a067524baf4f8521d5bb1cd0b8f')
+    version('2.2.2', 'd9f90a0d6e36df6c1312b6422280f2ec',
+            url = 'http://www.tddft.org/programs/octopus/down.php?file=libxc/libxc-2.2.2.tar.gz')
```
Comment (rmsds) [inline var/spack/repos/builtin/packages/libxc/package.py]:
```suggestion
            url='http://www.tddft.org/programs/octopus/down.php?file=libxc/libxc-2.2.2.tar.gz')
```

File: var/spack/repos/builtin/packages/libxc/package.py [hunk 2]
```diff
@@ -30,11 +30,21 @@ class Libxc(AutotoolsPackage):
...
+    version('2.2.1', '38dc3a067524baf4f8521d5bb1cd0b8f',
+            url = 'http://www.tddft.org/programs/octopus/down.php?file=libxc/libxc-2.2.1.tar.gz')
```
Comment (rmsds) [inline var/spack/repos/builtin/packages/libxc/package.py]:
```suggestion
            url='http://www.tddft.org/programs/octopus/down.php?file=libxc/libxc-2.2.1.tar.gz')
```

**Review by rmsds — CHANGES_REQUESTED** [review]
I think this is what is missing for flake8 to be happy.

**Review by rmsds — APPROVED** [review]

**Review by rmsds — APPROVED** [review]

**Comment by rmsds** [comment]
I think we need to change our CP2K spec in https://github.com/epfl-scitas/spack-packagelist to explicitly depend on `^libxc@3.0.0` before we merge this one. Otherwise a new CP2K 5.1 will be re-installed against the latest libxc: 4.2.3.

**Comment by rmsds** [comment]
Depends on https://github.com/epfl-scitas/spack-packagelist/pull/124

**Comment by alalazo** [comment]
@rmsds I updated the branch. Can you check if this is fine to be merged if all the tests go green?

**Review by rmsds — APPROVED** [review]
Looks good now, will merge once the updated checks go green.

**Pull request merged**